sign/rpmgensig.c

```diff
@@ -235,23 +235,26 @@ static int runGPG(sigTarget sigt, const char *sigfile)
     }
 
     if (!(pid = fork())) {
-	const char *tty = ttyname(STDIN_FILENO);
-	const char *gpg_path = NULL;
+	int using_gpg = (strstr(argv[0], "gpg") != NULL);
```

Comment:
This is a pretty lightweight check, but perhaps it is enough.  A more robust check might be greping the output of:

```
$ gpg --version
gpg (GnuPG-compatible Sequoia Chameleon) 2.2.40
Sequoia gpg Chameleon 0.11.0
sequoia-openpgp 1.21.2
...
$ /usr/bin/gpg --version
gpg (GnuPG) 2.2.40
libgcrypt 1.10.1
Copyright (C) 2022 g10 Code GmbH
...
```

Or:

```
$ gpg --list-config --with-colons version
cfg:version:2.2.40
```

Reply:
It's a feeble check for sure, added without much thought because autodetecting the command wasn't really on my radar to begin with. It's not hard to better though so we just as well might, thanks for pointing that out.